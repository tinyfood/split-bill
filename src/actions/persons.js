import uuid from 'uuid'

export const addPerson = (amount, locked = false) => ({
  type: 'ADD_PERSON',
  payload: {
    id: uuid(),
    amount,
    locked
  }
})

export const removePerson = ({ id } = {}) => ({
  type: 'REMOVE_PERSON',
  id
})

export const editPerson = (id, updates) => ({
  type: 'EDIT_PERSON',
  id,
  updates
})
