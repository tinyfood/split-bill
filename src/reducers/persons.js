const personsReducerDefaultState = [
  {
    id: 1,
    amount: 0,
    locked: false
  },
  {
    id: 2,
    amount: 0,
    locked: false
  },
  {
    id: 3,
    amount: 0,
    locked: false
  }
]

export default (state = personsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_PERSON':
      return [...state, action.payload]
    case 'REMOVE_PERSON':
      return state.filter(({ id }) => id !== action.id)
    case 'EDIT_PERSON':
      return state.map(person => {
        if (person.id === action.id) {
          return { ...person, ...action.updates }
        } else {
          return person
        }
      })
    default:
      return state
  }
}
