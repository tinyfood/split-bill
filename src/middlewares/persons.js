import { calculateTipsAmount, calculateTotal } from '../actions/split'

const personsMiddleware = ({ dispatch }) => next => action => {
  next(action)

  // Add action type check

  switch (action.type) {
    case 'EDIT_FRIENDS':
      alert('EDITED FRIENDS')
      break
    default:
      return
  }
}

export default personsMiddleware
