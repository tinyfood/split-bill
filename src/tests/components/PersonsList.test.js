import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { PersonsList } from '../../components/Personslist/Personslist'
import persons from '../fixtures/persons'

test('should render Personslist correctly', () => {
  const wrapper = shallow(<PersonsList persons={persons} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
