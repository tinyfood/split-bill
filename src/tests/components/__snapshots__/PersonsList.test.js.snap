// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Personslist correctly 1`] = `
<styled.ul>
  <Person
    amount={0}
    id={1}
    key="1"
  />
  <Person
    amount={0}
    id={2}
    key="2"
  />
  <Person
    amount={0}
    id={3}
    key="3"
  />
</styled.ul>
`;
